e2e/testcases/sync_ordering_test.go

```diff
@@ -516,12 +630,11 @@ func TestDependencyWithReconciliation(t *testing.T) {
 	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Add pod5 and pod6")
 	nt.WaitForRepoSyncs()
 
-	_, err = nomostest.Retry(30*time.Second, func() error {
-		var errs error
-		errs = multierr.Append(errs, nt.Validate("pod5", namespaceName, &corev1.Pod{}, isPodReady))
-		errs = multierr.Append(errs, nt.Validate("pod6", namespaceName, &corev1.Pod{}, isPodReady))
-		return errs
-	})
+	nt.T.Logf("Verify that pod5 and pod6 are ready")
+	err = multierr.Append(err, nt.Validate("pod5", namespaceName, &corev1.Pod{},
```

Comment:
Do we not need any sort of retry here? I assume it may not be `CurrentStatus` immediately after `WaitForRepoSyncs`

Reply:
With the explicit 5m reconcile timeout on this test, the pods _should_ all be current.

I increased the reconcile timeout for the whole test here to avoid needing to retry all the validations. 5m is the default timeout customers get anyway. 1m is just the e2e default.

Reply:
Where does the test wait for reconcile after adding `pod5`/`pod6`? My understanding is that `WaitForRepoSyncs` doesn't wait for the managed resources to reconcile

Reply:
Ah I just dug a bit. The applier in cli-utils waits for the specified ReconcileTimeout for the resources to be current?

Reply:
The applier DOES wait for reconcile, until the reconicle timeout, there's just no RSync status field that exposes the reconcile status, only sync status. And it won't update the Synced status until the applier has returned, after it's waited for the last apply wave to reconcile or timeout.

So WaitForRepoSyncs waits for reconcile until timeout, it just doesn't fail if there are reconcile failures, unless those are in dependencies that block syncing dependents.

Reply:
To really fix this UX, we need to add a Reconcile status condition to aggregate the status in ResourceGroup, and fix some bugs in the ResourceGroup reconcile status itself, which isn't being correctly updated by the resource-group-controller. Unfortunately, there's already a Reconciling condition that means something else (reconciler-manager updating the reconciler). So it might be a big change. I suspect we'll need it for upcoming UI enhancements and rollouts, but it hasn't been prioritized yet.